src/Diagnostics.RuntimeHost/Utilities/RequestBodyValidator.cs

```diff
@@ -20,5 +23,25 @@ public static bool ValidateRequestBody(JToken jsonBody, string[] fieldNames, out
             outputMessage = string.Concat(validationResults);
             return !(validationResults.Count() > 0);
         }
+
+        public static bool ValidateRequestBody(IRequestBodyBase body, out string outputMessage)
+        {
+            outputMessage = "";
+            if (body == null)
+            {
+                outputMessage = "Missing request body";
+                return false;
+            }
+
+            foreach (var property in body.GetType().GetProperties())
+            {
+                if (property.GetValue(body) == null)
+                {
+                    outputMessage = $"Missing {outputMessage += property.Name} from request body. ";
```

Comment:
This will generate a malformed message when multiple properties are missing. Don't do outputMessage += property.Name while assigning to outputMessage.
I would suggest to use a List object to store missing property names during the foreach loop, and create the outputMessage after the loop, like:

outputMessage = $"Missing {String.Join(", ", missingProperties)} from request body.";